src/qt/pivx/governancemodel.cpp

```diff
@@ -54,6 +54,7 @@ ProposalInfo GovernanceModel::buildProposalInfo(const CBudgetProposal* prop, boo
     // Calculate status
     int votesYes = prop->GetYeas();
     int votesNo = prop->GetNays();
+    int mnCount = mnodeman.CountEnabled();
```

Comment:
This loops over the masternodes maps for every existent proposal (legacy and deterministic). Locking the masternodes manager mutex on the GUI thread (which isn't good as other threads are accessing this mutex concurrently as well, which would freeze the GUI until the resource is available).

For the solution: we are already caching the masternodes count inside the `clientModel`(updating it on a timer thread), but we do it only as a string. This caches the int value: https://github.com/furszy/PIVX/commit/94aef020d5a643e1a3906654879a696b585d8714 

Feel free to squash it on your commit.

Reply:
yeah, this was one of the follow-ups i had in mind, but i suppose it's ok to include it here directly....so, done